store/src/meta_service/state_machine.rs

```diff
@@ -550,6 +550,22 @@ impl StateMachine {
                 tracing::debug!("applied UpsertKV: {} {:?}", key, record_value);
                 Ok((prev, record_value).into())
             }
+
+            Cmd::TruncateTable {
+                ref db_name,
+                ref table_name,
+            } => {
+                let db = self.databases.get_mut(db_name).unwrap();
+                let tbl_id = db.tables.get(table_name);
+                if let Some(tbl_id) = tbl_id {
+                    let tbl_id = tbl_id.to_owned();
+                    self.remove_table_data_parts(db_name, table_name);
+                    let prev = self.tables.get(&tbl_id);
+                    Ok((prev.cloned(), prev.cloned()).into())
+                } else {
+                    Ok((None::<Table>, None::<Table>).into())
+                }
+            }
```

Comment:
The two fields `prev`, `result` in  `AppliedState` implies the state before and after applying an `Cmd`.
For `truncate table`, no matter whether it succeeded, the state of the table never changes.
What changed is actually the table parts.

So, here the `AppliedState` should reflect the table parts changes instead of the table changes.

Since the `AppliedState` is only used internally for now, and it is inappropriate to put the entire list of parts into `AppliedState`, my suggestion is to introduce a struct to describe the table parts change. 

BTW. there should be a corresponding `seq` for a `table`, as it is for `UpsertKV`.
Then we could use the `seq` number to identify a table change.
Maybe another issue 🤔

Reply:
thanks, a new struct for `AppliedState` is great idea, and i will add seq for table to identify a table change in this pr, but this seq only for debug?

Reply:
@SGZW not for debug. It is used to identify a unique version of some data. Normally the caller just checks the `seq` to learn if the `table` is change. And maybe `seq` is used to invalidate caches.

Reply:
i add seq for table in other pr